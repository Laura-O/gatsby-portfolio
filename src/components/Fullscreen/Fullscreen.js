import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import * as constants from '../../layouts/shared/style-constants';

const FullscreenWrapper = styled.div`
  width: 100vw;
  height: 100vh;
  max-width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

  ${props => {
    if (props.active) {
      return `
      background: ${constants.darkBackground};
      color: ${constants.colorWhite};
      position: fixed;
      top: 0;
      left: 0;
      right: 0;
      z-index: 5;
      opacity: 0.99;
      `;
    }
    return `
        display: none;
      `;
  }};
`;

const Fullscreen = ({ children, active }) => (
  // const containerClassName = classNames(styles.fullscreen, className);

  <FullscreenWrapper active={active}>{children}</FullscreenWrapper>
);

export default Fullscreen;

Fullscreen.propTypes = {
  children: PropTypes.node,
  active: PropTypes.bool.isRequired,
};
