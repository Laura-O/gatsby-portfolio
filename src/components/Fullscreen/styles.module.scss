@import '../../scss/variables.scss';

.fullscreen {
  width: 100vw;
  height: 100vh;
  max-width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.first {
  margin-top: calc($headerHeight * -1);
  z-index: 999;
}
