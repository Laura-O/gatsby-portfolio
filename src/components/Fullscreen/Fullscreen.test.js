import React from 'react';
import { mount } from 'enzyme';
import Fullscreen from './Fullscreen';

describe('fullscreen mode', () => {
  test('active', () => {
    const wrapper = mount(<Fullscreen active={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('active', () => {
    const wrapper = mount(<Fullscreen active={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
