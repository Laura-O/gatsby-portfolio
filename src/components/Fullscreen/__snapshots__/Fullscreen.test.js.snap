// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fullscreen mode active 1`] = `
<Fullscreen
  active={true}
>
  <styled.div
    active={true}
  >
    <div
      className="sc-bdVaJa jcHNws"
    />
  </styled.div>
</Fullscreen>
`;

exports[`fullscreen mode active 2`] = `
<Fullscreen
  active={false}
>
  <styled.div
    active={false}
  >
    <div
      className="sc-bdVaJa hxjTWr"
    />
  </styled.div>
</Fullscreen>
`;
